examples/function-pointers/function_pointers.bal

```diff
@@ -1,44 +1,24 @@
 import ballerina/io;
-import ballerina/lang.'int;
 
-public function main() {
-    // The function name `test` serves as a function pointer argument in the
-    // call to `foo()`. Function names can be thought of as final variables
-    // since although you can use it like a regular variable, you cannot
-    // modify the value it is associated with.
-    io:println("Answer: ", foo(10, test));
-    io:println("Answer: ", foo(10, getFunctionPointer()));
-
-    // A function pointer as a variable.
-    function (string, int...) returns float f = getFunctionPointer();
+function add(int v1, int v2) returns int {
+    return v1 + v2;
+}
 
-    io:println("Answer: ", foo(10, f));
+function multiply(int v1, int v2) returns int {
+    return v1 * v2;
 }
 
 // A function pointer as a parameter. A function pointer can be invoked similar
 // to how a normal function is invoked.
-function foo(int x, function (string, int...) returns float bar)
-             returns float {
-    return x * bar("2", 2, 3, 4, 5);
+function process(function (int, int) returns int func, int v1, int v2) returns int {
+    return func(v1, v2);
 }
 
-// A function type as the return type.
-function getFunctionPointer() returns
-                    (function (string, int...) returns float) {
-    // Returning a function pointer.
-    return test;
-}
-
-function test(string s, int... x) returns float {
-    int|error y = 'int:fromString(s);
-    float f = 0.0;
-
-    if (y is int) {
-        foreach var item in x {
-            f += item * 1.0 * y;
-        }
-    } else {
-        panic y;
-    }
-    return f;
+public function main() {
+    // The function name `add` serves as a function pointer argument in the
+    // call to `process()`. Function names can be thought of as final variables
+    // since although you can use it like a regular variable, you cannot
+    // modify the value it is associated with.
```

Comment:
```suggestion
    // modify the values they are associated with.
```

Reply:
The value we are referring to here is the function, where we say it is like a regular variable, so it's a single value we talk about there.